recommender/engines/autoencoders/training/model_training_step.py

```diff
@@ -104,16 +105,24 @@ def autoencoder_loss_function(reconstructions, features):
     return cos_emb_loss(reconstructions, features, ones)
 
 
-def evaluate_autoencoder(model, dataloader, loss_function, device):
+def evaluate_autoencoder(
+    model: AutoEncoder,
+    dataloader: DataLoader,
+    loss_function: Callable[[Tensor, Tensor], Tensor],
+    acc_function: Callable[[Tensor, Tensor, bool], float],
+    device: torch.device = torch.device("cpu"),
+) -> Tuple[float, float]:
     """Evaluate autoencoder"""
-
+    model = model.to(device)
     model.eval()
     with torch.no_grad():
         for batch in dataloader:
             features = batch[0].float().to(device)
-            preds = model(features).to(device)
-            loss = loss_function(features, preds)
-        return loss.item()
+            reconstructions = model(features).to(device)
```

Comment:
Can't we use the new `get_reconstructions` function here?

Reply:
In eval, we could even skip splitting the test dataset to batches and just evaluate the whole dataset but that may require mroe changes elsewhere

Reply:
This reminds me, we should consider including `mypy` type checker into the githooks/CI